### PR TITLE
Issue/5955 reader comment count

### LINKED ISSUE #5955: Reader comment count inaccurate
When viewing the reader comment list, the comment count isn't always correct. The post should be updated when this activity is opened to ensure we have the latest comment count.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #5955 - prior to this PR, the reader comment list wouldn't update the associated post. This sometimes led to incorrect comment & like counts, since those numbers may have changed since the post was last requested.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java [hunk 1]
```diff
@@ -189,43 +183,38 @@ public void onClick(View v) {
...
+                    if (post != null) {
+                        getCommentAdapter().setPost(post);
+                        mPost = post;
+                    }
                 }
             }
         });
+
+        // clear all the previous comments, then load the first page of comments
+        ReaderCommentTable.purgeCommentsForPost(mBlogId, mPostId);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java]: Are we really sure we want to purge all the comments here before waiting for server response? 
If the network call fails the client is stuck with no comments at all. I just tried by making the call failing, and then rotated the device.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java]: @daniloercoli While looking into this I discovered another issue with the incorrect comment count being shown. I'm going to put this PR on hold until I have time to track down that problem.

**Comment by nbradbury** [comment]
@danilo This is ready for review now. I took your advice and moved purging comment until after the response in a414504. The other change is in c7a4bf7, where I make sure all instances of the updated post reflect the latest comment count.

**Comment by daniloercoli** [comment]
Looks good! :shipit:

**Pull request merged**